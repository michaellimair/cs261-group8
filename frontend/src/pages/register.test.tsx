import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import RegisterPage from './register';

describe('register', () => {
  it('renders properly', () => {
    const result = render(
      <BrowserRouter>
        <RegisterPage />
      </BrowserRouter>,
    );
    expect(result).toMatchSnapshot();
  });

  it('goes to login page when login button is clicked', async () => {
    const result = render(
      <BrowserRouter>
        <RegisterPage />
      </BrowserRouter>,
    );
    const registerButton = await result.findByText('Login');
    registerButton.click();
    expect(window.location.pathname).toBe('/auth');
  });
});
