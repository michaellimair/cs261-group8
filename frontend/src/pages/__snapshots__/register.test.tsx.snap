// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-owjkmg"
      >
        <h2
          class="chakra-heading css-5efzzv"
        >
          register
        </h2>
      </div>
      <div
        class="css-3hziw7"
      >
        <div
          class="chakra-stack css-17kdbol"
        >
          <form
            data-testid="registerForm"
          >
            <div
              class="chakra-stack css-1u8k0pa"
            >
              <div
                class="css-0"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="firstName"
                    id="firstName-label"
                  >
                    firstName
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <input
                    aria-required="true"
                    autocomplete="given-name"
                    class="chakra-input css-0"
                    data-testid="first_name"
                    id="firstName"
                    name="first_name"
                    required=""
                    type="text"
                  />
                </div>
              </div>
              <div
                class="css-0"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="lastName"
                    id="lastName-label"
                  >
                    lastName
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <input
                    aria-required="true"
                    autocomplete="given-name"
                    class="chakra-input css-0"
                    data-testid="last_name"
                    id="lastName"
                    name="last_name"
                    required=""
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-wlea3r"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="username"
                id="username-label"
              >
                username
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                autocomplete="username"
                class="chakra-input css-0"
                data-testid="username"
                id="username"
                name="username"
                required=""
                type="text"
              />
            </div>
            <div
              class="chakra-form-control css-wlea3r"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="email"
                id="email-label"
              >
                email
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                autocomplete="email"
                class="chakra-input css-0"
                data-testid="email"
                id="email"
                name="email"
                required=""
                type="email"
              />
            </div>
            <div
              class="chakra-form-control css-wlea3r"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="new-password"
                id="new-password-label"
              >
                password
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="chakra-input__group css-4302v8"
              >
                <input
                  aria-required="true"
                  autocomplete="new-password"
                  class="chakra-input css-0"
                  data-testid="password"
                  id="new-password"
                  name="password"
                  required=""
                  type="password"
                />
                <div
                  class="chakra-input__right-element css-1aezdzf"
                >
                  <button
                    class="chakra-button css-5c2176"
                    data-testid="show-password-button"
                    type="button"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      data-testid="password-hidden"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        fill="currentColor"
                      >
                        <path
                          d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                        />
                        <path
                          d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                        />
                      </g>
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-wlea3r"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="verify-password"
                id="verify-password-label"
              >
                verify_password
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="chakra-input__group css-4302v8"
              >
                <input
                  aria-required="true"
                  autocomplete="new-password"
                  class="chakra-input css-0"
                  data-testid="verify_password"
                  id="verify-password"
                  name="verify_password"
                  required=""
                  type="password"
                />
              </div>
            </div>
            <div
              class="chakra-stack css-1mwdmne"
            >
              <button
                class="chakra-button css-18dry65"
                data-testid="registerButton"
                type="submit"
              >
                register
              </button>
              <div
                class="chakra-form-control css-t9xfd"
                role="group"
              />
            </div>
            <div
              class="chakra-stack css-nkdhe4"
            >
              <p
                class="chakra-text css-xi606m"
              >
                already_a_user
                 
                <a
                  class="chakra-link css-p03q1r"
                  data-testid="login-button"
                  href="/auth"
                >
                  login
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-owjkmg"
    >
      <h2
        class="chakra-heading css-5efzzv"
      >
        register
      </h2>
    </div>
    <div
      class="css-3hziw7"
    >
      <div
        class="chakra-stack css-17kdbol"
      >
        <form
          data-testid="registerForm"
        >
          <div
            class="chakra-stack css-1u8k0pa"
          >
            <div
              class="css-0"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="firstName"
                  id="firstName-label"
                >
                  firstName
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  autocomplete="given-name"
                  class="chakra-input css-0"
                  data-testid="first_name"
                  id="firstName"
                  name="first_name"
                  required=""
                  type="text"
                />
              </div>
            </div>
            <div
              class="css-0"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="lastName"
                  id="lastName-label"
                >
                  lastName
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  autocomplete="given-name"
                  class="chakra-input css-0"
                  data-testid="last_name"
                  id="lastName"
                  name="last_name"
                  required=""
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-form-control css-wlea3r"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="username"
              id="username-label"
            >
              username
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <input
              aria-required="true"
              autocomplete="username"
              class="chakra-input css-0"
              data-testid="username"
              id="username"
              name="username"
              required=""
              type="text"
            />
          </div>
          <div
            class="chakra-form-control css-wlea3r"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="email"
              id="email-label"
            >
              email
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <input
              aria-required="true"
              autocomplete="email"
              class="chakra-input css-0"
              data-testid="email"
              id="email"
              name="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="chakra-form-control css-wlea3r"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="new-password"
              id="new-password-label"
            >
              password
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="chakra-input__group css-4302v8"
            >
              <input
                aria-required="true"
                autocomplete="new-password"
                class="chakra-input css-0"
                data-testid="password"
                id="new-password"
                name="password"
                required=""
                type="password"
              />
              <div
                class="chakra-input__right-element css-1aezdzf"
              >
                <button
                  class="chakra-button css-5c2176"
                  data-testid="show-password-button"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-onkibi"
                    data-testid="password-hidden"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                      />
                      <path
                        d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="chakra-form-control css-wlea3r"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="verify-password"
              id="verify-password-label"
            >
              verify_password
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="chakra-input__group css-4302v8"
            >
              <input
                aria-required="true"
                autocomplete="new-password"
                class="chakra-input css-0"
                data-testid="verify_password"
                id="verify-password"
                name="verify_password"
                required=""
                type="password"
              />
            </div>
          </div>
          <div
            class="chakra-stack css-1mwdmne"
          >
            <button
              class="chakra-button css-18dry65"
              data-testid="registerButton"
              type="submit"
            >
              register
            </button>
            <div
              class="chakra-form-control css-t9xfd"
              role="group"
            />
          </div>
          <div
            class="chakra-stack css-nkdhe4"
          >
            <p
              class="chakra-text css-xi606m"
            >
              already_a_user
               
              <a
                class="chakra-link css-p03q1r"
                data-testid="login-button"
                href="/auth"
              >
                login
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
